### PR TITLE
Migrate `use-suggestions-rust-2018`, `overwrite-input`, `lto-dylib-dep` and `many-crates-but-no-match` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

### CODE REVIEW

File: tests/run-make/msvc-opt-minsize/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// A previously outdated version of LLVM caused compilation failures on Windows
+// specifically with optimization level `z`. After the update to a more recent LLVM
+// version, this test checks that compilation and execution both succeed.
+// See https://github.com/rust-lang/rust/issues/45034
+
+// NOTE: This isn't an UI test despite its simplicity, because UI tests
+// cannot contain opt-level modifications.
```
Comment (Oneirical) [inline tests/run-make/msvc-opt-minsize/rmake.rs:7]: I'm not so sure about that. It's what was said [here](https://github.com/rust-lang/rust/pull/45824#issuecomment-343591668), but that was 7 years ago and I definitely see some instances of opt-level in the UI testsuite. Let me know if you think I should attempt a conversion to an UI test.

Comment (Kobzol) [reply tests/run-make/msvc-opt-minsize/rmake.rs:7]: Yeah this really looks like it should be a UI test, they can set `-Copt-level` flags now.

**Review by Kobzol — COMMENTED** [review]
Looks good! Since this already contains 5 tests, I'd move the UI test port to a different PR, and we can try to merge this one without `msvc-opt-minsize`.

**Comment by Oneirical** [comment]
> Looks good! Since this already contains 5 tests, I'd move the UI test port to a different PR, and we can try to merge this one without `msvc-opt-minsize`.

I just dropped it from this PR.

**Comment by Kobzol** [comment]
Okay, here goes nothing.

@bors r+ rollup=iffy

**Pull request merged**